### ISSUE TITLE
Wrong EfficientAD number_of_steps for optimizer lr change?

### ISSUE THREAD

**Issue by AlessandroPolidori** [openvinotoolkit/anomalib#1234]
https://github.com/openvinotoolkit/anomalib/blob/784767fc2f19a8f354f152aba7f4338cb628118c/src/anomalib/models/efficient_ad/lightning_model.py#L207C21-L207C24

Shouldn't this be the minimum between the two?

**Comment by AlessandroPolidori** [comment]
Also, @nelson1425 @alexriedel1 i'm having trouble to obtain results as high as the ones i get with the nelson1425 original efficientAD repo. I'm racking my brain trying to figure out what could be wrong and the most plausible thing is that here in anomalib when the transform is none it still does the albumentations.ToFloat (which not only translates to float but also normalizes the values) and then at the start of the forward the batch gets normalized again using the imagenet mean and std. Have you also noticed a performance drop passing from nelson repo to the anomalib refactoring?

**Comment by alexriedel1** [comment]
> Also, @nelson1425 @alexriedel1 i'm having trouble to obtain results as high as the ones i get with the nelson1425 original efficientAD repo. I'm racking my brain trying to figure out what could be wrong but the only substantial difference i found is the normalization of the images. Here in anomalib the dataloader applies albumentations.ToFloat and then the normalization wrt imagenet's mean and std happens at the start of the forward, while in nelson's repo you directly apply albumentations normalization. Have you also noticed a performance drop passing from nelson repo to the anomalib refactoring?

As @nelson1425 is using torchvision transforms in general and anomalib uses albumentations for image augmentations, there are more differences than just the normalizu process. Also the resizing process doesn't use the exact same method to obtain the same results. This can be solved if anomalib switches to torchvision transforms or if you train a teacher model on imagenet using albumentation transforms I guess.

**Comment by samet-akcay** [comment]
We will be switching to torchvision, but we plan to do so after releasing v1.0, hopefully in October

**Comment by samet-akcay** [comment]
> https://github.com/openvinotoolkit/anomalib/blob/784767fc2f19a8f354f152aba7f4338cb628118c/src/anomalib/models/efficient_ad/lightning_model.py#L207C21-L207C24
> 
> Shouldn't this be the minimum between the two?

@alexriedel1 would you like to reply to this?

**Comment by alexriedel1** [comment]
> https://github.com/openvinotoolkit/anomalib/blob/784767fc2f19a8f354f152aba7f4338cb628118c/src/anomalib/models/efficient_ad/lightning_model.py#L207C21-L207C24
> 
> Shouldn't this be the minimum between the two?

Yes should be min. Do you mind creating a PR for this?

**Comment by samet-akcay** [comment]
Thanks @alexriedel1. @AlessandroPolidori, you are more than welcome to create a PR so you could become one of our contributors
